src/pushsource/_impl/backend/errata_source/errata_client.pyi

```diff
@@ -0,0 +1,32 @@
+# TODO: are stub files subject to the same need for six.moves?
+import xmlrpc.client
+from collections import Sequence
+from typing import Any, TypeVar
+
+from pushsource.type_aliases import JsonObject
+
+ErrataRaw_co = TypeVar("ErrataRaw_co", bound="ErrataRaw", covariant=True)
+
+class ErrataRaw(object):
+    advisory_cdn_metadata: JsonObject
+    advisory_cdn_file_list: JsonObject
+    advisory_cdn_docker_file_list: JsonObject
+    ftp_paths: JsonObject
+    def __init__(
+        self,
+        advisory_cdn_metadata: JsonObject,
+        advisory_cdn_file_list: JsonObject,
+        advisory_cdn_docker_file_list: JsonObject,
+        ftp_paths: JsonObject,
+    ) -> None: ...
+
+class ErrataClient(object):
+    _errata_service: xmlrpc.client.ServerProxy
+    def __init__(self, threads: int, url: str, **retry_args: Any) -> None: ...
+    def shutdown(self) -> None: ...
+    def _log_queried_et(self, response: JsonObject, advisory_id: str) -> JsonObject: ...
```

Comment:
> I'm curious does it cause problems if some of these are left out?

It doesn't cause problems to leave things out from the stub files in the sense that there's no issue at runtime. However, it is worth noting that IDEs will likely handle this in different ways. Using PyCharm as an example, it emits an unresolved reference Warning for `_log_queried_et` if I take it out of the stub file (see first screenshot), and flags imports of names missing from stubs as an error (see second screenshot). The severity of these happens to be configurable in PyCharm, but that may not be the case in all IDEs.

![image](https://user-images.githubusercontent.com/16199909/149163237-62c96f86-d271-4dfe-8a8a-89b439ea235b.png)

![image](https://user-images.githubusercontent.com/16199909/149163129-b4f02a45-d236-47f7-b9eb-7b2c64f505dd.png)

In the first case the warning is easily remedied by leaving the function signature in the stub file, but removing the type hints from it, which could be a happy middle ground. This would look like:

```python
def _log_queried_et(self, response, advisory_id): ...
```

This approach would attempt to keep the type information private to anyone who reads the stub file, but also prevents the unresolved reference warning.

> I don't want to require that type hints must be added even for private methods on private classes

I agree with this sentiment, and it would be worth documenting that type hints are not required in the README or published library docs for contributing to the library that this is the case.